src/Microsoft.VisualStudio.SDK.Analyzers/VSSDK006CheckServicesExistAnalyzer.cs

```diff
@@ -184,14 +184,22 @@ where symbol.Equals(symbolAccessed)
             private bool IsNonNullCheck(SyntaxNode node, ISymbol symbol, SyntaxNodeAnalysisContext context)
             {
                 bool IsSymbol(SyntaxNode n) => symbol.Equals(context.SemanticModel.GetSymbolInfo(n, context.CancellationToken).Symbol);
-
-                if (node is IfStatementSyntax ifStatement &&
-                    ifStatement.Condition.DescendantNodesAndSelf().OfType<BinaryExpressionSyntax>().Any(
-                        o => (o.OperatorToken.IsKind(SyntaxKind.EqualsEqualsToken) || o.OperatorToken.IsKind(SyntaxKind.ExclamationEqualsToken))
-                            && (o.Left.IsKind(SyntaxKind.NullLiteralExpression) || o.Right.IsKind(SyntaxKind.NullLiteralExpression))
-                            && (IsSymbol(o.Left) || IsSymbol(o.Right))))
+                bool IsEqualsOrExclamationEqualsCheck(BinaryExpressionSyntax o) => (o.OperatorToken.IsKind(SyntaxKind.EqualsEqualsToken) || o.OperatorToken.IsKind(SyntaxKind.ExclamationEqualsToken))
```

Comment:
Can you try adding `static` modifiers to these local functions to ensure they don't pick up closures and add more allocations to analyzing?

Reply:
static local functions is a C# 8 feature :(

Also, these can't be static, as they depend on `IsSymbol`, which has closures on `symbol` and `context`.  However, if I'm looking at the ildasm (and understand the compiler correctly), these all end up on the same object so it's the same number of allocations, but slightly larger to accommodate the new methods.

![image](https://user-images.githubusercontent.com/3976973/78108347-0beef400-73ac-11ea-9df9-8293192efc8d.png)

Reply:
Oh! That's great that C# optimizes that.
We can certainly bump the lang version to 8 if that helped, but I'm satisfied with your answer.